code/game/machinery/door_control.dm

```diff
@@ -153,7 +153,7 @@
 				if(specialfunctions & BOLTS)
 					D.lock()
 				if(specialfunctions & SHOCK)
-					D.secondsElectrified = -1
+					D.electrified = -1
```

Comment:
How about a `#define PERMANENT -1` or something along `DOESN'T RESET` for code clarity ?